lib/hyacinth/digital_object/resource.rb

```diff
@@ -45,6 +45,29 @@ def to_serialized_form
       def as_json(_options = {})
         SERIALIZED_FIELDS.map { |field| [field.to_s, self.send(field)] }.to_h
       end
+
+      def image?
+        BestType.pcdm_type.for_mime_type(media_type) == 'Image' ||
+          BestType.pcdm_type.for_file_name(original_filename) == 'Image'
+      end
+
+      def video?
+        BestType.pcdm_type.for_mime_type(media_type) == 'Video' ||
+          BestType.pcdm_type.for_file_name(original_filename) == 'Video'
+      end
+
+      def audio?
+        BestType.pcdm_type.for_mime_type(media_type) == 'Audio' ||
+          BestType.pcdm_type.for_file_name(original_filename) == 'Audio'
+      end
+
+      def pdf?
+        media_type == 'application/pdf' || original_filename.ends_with?('.pdf')
+      end
+
+      def text_or_office_document?
+        media_type.match?(/text|msword|ms-word|officedocument|powerpoint|excel|iwork/)
```

Comment:
could we get this from BestType returned values?

Reply:
Good idea!  I checked the BestType pcdm type lookup rules and I'll need to make a couple of minor updates to fully cover the cases here, but that's not hard to do.  I'll update this code now to use BestType and then separately update BestType.